jupyter-lib/api/src/main/kotlin/org/jetbrains/kotlinx/jupyter/api/libraries/connection.kt

```diff
@@ -0,0 +1,190 @@
+package org.jetbrains.kotlinx.jupyter.api.libraries
+
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonObject
+import org.jetbrains.kotlinx.jupyter.util.EMPTY
+
+/**
+ * Jupyter connection socket types
+ * Here you can find an information about Jupyter sockets:
+ * https://jupyter-client.readthedocs.io/en/stable/messaging.html#introduction
+ *
+ * For now, only adding callbacks for messages on `control` and `shell` sockets makes sense.
+ */
+enum class JupyterSocket {
```

Comment:
Looks like this enum provides same info as `JupyterSocketInfo`, perhaps, it's better to incorporate one into another

Reply:
`JupyterSocketInfo` has some implementation details which I don't want to expose to the public API. I've incorporated one into another however